inc/routing_table.h

```diff
@@ -93,4 +103,9 @@ uint16_t RoutingTB_GetLastService(void);
 uint16_t *RoutingTB_GetLastNode(void);
 uint16_t RoutingTB_GetLastEntry(void);
 
+// ********************* research routing table ********************************
+search_result_t *RTFilter_Type(search_result_t *result, luos_type_t type);
+search_result_t *RTFilter_Node(search_result_t *result, uint16_t node_id);
+uint16_t RTFilter_ExtarctServiceID(search_result_t *result, uint16_t *id_table);
```

Comment:
Following my previous feedback,
Why users just doesn't do:
```C
result. result_table[0].id;
```

Reply:
they can, but that is to say users need to know how searching_result_t as well as routing_table_t work.

Reply:
They will copy-paste it.
If the line is clear they will get it easily.
To make it smoother you could replace the result_table as table or list to be able to make
```C
result. list[0].id;
```

Reply:
but they will need to check the mode of this entry in the list. if it is a service or a node and which id they want. For example when we do RTFilter_Type() we keep all the services of the needed type and all the entries for their nodes in order to be able to understand in which node they are... The user should apply a loop and search by mode = SERVICE to find their ids...

Reply:
I don't think your research function will have to return node entry. only services.